src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -103,9 +110,15 @@ def get_user_agent(self):
         """
 
         def parts():
-            if "user_agent" in self:
-                yield self.get("user_agent")
-            yield f"promptflow/{VERSION}"
+            # Be careful not to use "self.user_agent" as it will recursively call get_user_agent in __getattr__
+            agent = self.get(self._USER_AGENT, '')
+            yield agent
+            promptflow_agent = f"promptflow/{VERSION}"
+            yield promptflow_agent if promptflow_agent not in agent else ''
+            user_agent = os.environ.get(USER_AGENT, '').strip()
+            yield user_agent if user_agent not in agent else ''
+            pf_user_agent = os.environ.get(PF_USER_AGENT, '').strip()
+            yield pf_user_agent if pf_user_agent not in agent else ''
```

Comment:
Maybe we need to handle case when there are multiple user agents in env var, e.g.: "ua-1/1.0.0 ua-2/1.0.0"

Reply:
Do you mean for the case of **"ua-1/1.0.0 ua-2/1.0.0"**, where "ua-1/1.0.0" already has an agent, but **"ua-2/1.0.0"** does not exist, so only **"ua-2/1.0.0"** needs to be added instead of the entire **"ua-1/1.0.0 ua-2/1.0.0"** agent?